perception/sub8_perception/nodes/path_marker_finder.py

```diff
@@ -63,20 +63,36 @@ def __init__(self):
         self.last_found_time = None
 
         # Constants from launch config file
-        self.debug_ros = rospy.get_param("debug_ros", True)
-        self.canny_low = rospy.get_param("canny_low", 100)
-        self.canny_ratio = rospy.get_param("canny_ratio", 3.0)
-        self.thresh_hue_high = rospy.get_param("thresh_hue_high", 60)
-        self.thresh_saturation_low = rospy.get_param("thresh_satuation_low", 100)
-        self.min_contour_area = rospy.get_param("min_contour_area", 100)
-        self.length_width_ratio_err = rospy.get_param("length_width_ratio_err", 0.2)
-        self.approx_polygon_thresh = rospy.get_param("approx_polygon_thresh", 10)
-        self.shape_match_thresh = rospy.get_param("shape_match_thresh", 0.4)
-        camera = rospy.get_param("marker_camera", "/camera/down/left/image_rect_color")
+        self.debug_ros = rospy.get_param("~debug_ros", True)
+        self.canny_low = rospy.get_param("~canny_low", 100)
+        self.canny_ratio = rospy.get_param("~canny_ratio", 3.0)
+        self.thresh_hue_high = rospy.get_param("~thresh_hue_high", 60)
+        self.thresh_saturation_low = rospy.get_param("~thresh_satuation_low", 100)
+        self.min_contour_area = rospy.get_param("~min_contour_area", 100)
+        self.length_width_ratio_err = rospy.get_param("~length_width_ratio_err", 0.2)
+        self.approx_polygon_thresh = rospy.get_param("~approx_polygon_thresh", 10)
+        self.shape_match_thresh = rospy.get_param("~shape_match_thresh", 0.4)
+        self.min_found_count = rospy.get_param("~min_found_count", 10)
+        self.timeout_seconds = rospy.get_param("~timeout_seconds", 2.0)
+        camera = rospy.get_param("~marker_camera", "/camera/down/left/image_rect_color")
+
+        self.state_size = 8 # X, Y, Z, THETA, vx, vy, vz, vtheta
```

Comment:
As I'm measuring the pose in the camera frame, it will move over time. Honestly I don't really know what I'm doing but other people online seem to track velocity for computer vision object tracking